roachtest: backup/mixed-version failed

roachtest.backup/mixed-version [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3924331&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3924331&tab=artifacts#/backup/mixed-version) on release-21.2 @ [3adc6da4ccdf80618e6bbcee5ee9e503a92517e7](https://github.com/cockroachdb/cockroach/commits/3adc6da4ccdf80618e6bbcee5ee9e503a92517e7):


```
The test failed on branch=release-21.2, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/backup/mixed-version/run_1
	test_impl.go:274,assertions.go:262,assertions.go:1332,require.go:1231,backup.go:196,versionupgrade.go:207,backup.go:221,test_runner.go:777: 
			Error Trace:	backup.go:196
			            				versionupgrade.go:207
			            				backup.go:221
			            				test_runner.go:777
			            				asm_amd64.s:1581
			Error:      	Received unexpected error:
			            	dial tcp 35.229.38.66:26257: connect: connection refused
			Test:       	backup/mixed-version

	cluster.go:1296,context.go:89,cluster.go:1284,test_runner.go:866: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-3924331-1639639203-10-n4cpu4 --oneshot --ignore-empty-nodes: exit status 1 4: 10323
		3: 10810
		2: dead (exit status 134)
		1: 11451
		Error: UNCLASSIFIED_PROBLEM: 2: dead (exit status 134)
		(1) UNCLASSIFIED_PROBLEM
		Wraps: (2) attached stack trace
		  -- stack trace:
		  | github.com/cockroachdb/cockroach/pkg/roachprod.Monitor
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/roachprod/roachprod.go:596
		  | main.glob..func14
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:569
		  | main.wrap.func1
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:123
		  | github.com/spf13/cobra.(*Command).execute
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:856
		  | github.com/spf13/cobra.(*Command).ExecuteC
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:960
		  | github.com/spf13/cobra.(*Command).Execute
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:897
		  | main.main
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:1170
		  | runtime.main
		  | 	/usr/local/go/src/runtime/proc.go:255
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
		Wraps: (3) 2: dead (exit status 134)
		Error types: (1) errors.Unclassified (2) *withstack.withStack (3) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73708 roachtest: backup/mixed-version failed [C-test-failure O-roachtest O-robot T-bulkio branch-master release-blocker]
- #73327 roachtest: backup/mixed-version failed [C-test-failure O-roachtest O-robot T-bulkio branch-release-21.1]
</p>
</details>
/cc @cockroachdb/bulk-io
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*backup/mixed-version.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cockroach-stderr.log on node 2 indications some sort seg fault in the stopper function, with the following stack trace:
```
panic: runtime error: invalid memory address or nil pointer dereference          
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=**0x10b5645**]         
                                                                                 
goroutine 596 [running]:                                                         
panic({0x4246540, 0xab136d0})                                                    
        /usr/local/go/src/runtime/panic.go:1147 +0x3a8 fp=0xc002a4ddc0 sp=0xc002a4dd00 pc=0x488428
runtime.panicmem(...)                                                            
        /usr/local/go/src/runtime/panic.go:221                                   
runtime.sigpanic()                                                               
        /usr/local/go/src/runtime/signal_unix.go:735 +0x327 fp=0xc002a4de10 sp=0xc002a4ddc0 pc=0x49ee47
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).WithCancelOnQuiesce(0x0, {0x815b2c0, 0xc000076088})
        /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:271 +0x85 fp=0xc002a4dea0 sp=0xc002a4de10 pc=**0x10b5645**
github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptcache.(*Cache).doSingleFlightUpdate(0xc000e03130)
        /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptcache/cache.go:191 +0x57 fp=0xc002a4df20 sp=0xc002a4dea0 pc=0x31b9cf7
github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptcache.(*Cache).doSingleFlightUpdate-fm()
        /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/protectedts/ptcache/cache.go:189 +0x26 fp=0xc002a4df38 sp=0xc002a4df20 pc=0x31baae6
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).doCall(0xc000e03158, 0xc001d9bae0, {0x0, 0x0}, 0xc001778790)
        /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:128 +0x3b fp=0xc002a4dfa8 sp=0xc002a4df38 pc=0x18e241b
github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan·dwrap·1()
        /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:121 +0x36 fp=0xc002a4dfe0 sp=0xc002a4dfa8 pc=0x18e23b6
runtime.goexit()                                                                 
        /usr/local/go/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc002a4dfe8 sp=0xc002a4dfe0 pc=0x4bc2a1
created by github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight.(*Group).DoChan
        /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight/singleflight.go:121 +0x2f1
```

--- a/pkg/server/server.go
+++ b/pkg/server/server.go
@@ -1671,6 +1671,19 @@ func (s *Server) PreStart(ctx context.Context) error {
 		}
 	})
 
+	// Start the protected timestamp subsystem. Note that this needs to happen
+	// before the modeOperational switch below, as the protected timestamps
+	// subsystem will crash if accessed before being Started (and serving general
+	// traffic may access it).
+	//
+	// See https://github.com/cockroachdb/cockroach/issues/73897.
+	if err := s.protectedtsProvider.Start(ctx, s.stopper); err != nil {
+		return err
+	}
+	if err := s.protectedtsReconciler.Start(ctx, s.stopper); err != nil {
+		return err
+	}
+
 	// After setting modeOperational, we can block until all stores are fully
 	// initialized.
 	s.grpc.setMode(modeOperational)
@@ -1714,14 +1727,6 @@ func (s *Server) PreStart(ctx context.Context) error {
 
 	// Begin recording status summaries.
 	if err := s.node.startWriteNodeStatus(base.DefaultMetricsSampleInterval); err != nil {
-		return err
-	}
-
-	// Start the protected timestamp subsystem.
-	if err := s.protectedtsProvider.Start(ctx, s.stopper); err != nil {
-		return err
-	}
-	if err := s.protectedtsReconciler.Start(ctx, s.stopper); err != nil {
 		return err
 	}
 

